Fixes compilation of the curses module on Cygwin; see
https://bugs.python.org/issue14438

This patch was never accepted upstream, though there is a new fix for it
in progress at https://bugs.python.org/issue28190 but also not yet accepted.
--- a/Include/py_curses.h	2009-09-06 16:23:05.000000000 -0500
+++ b/Include/py_curses.h	2010-04-14 15:21:23.008971400 -0500
@@ -17,6 +17,13 @@
 #define NCURSES_OPAQUE 0
 #endif /* __APPLE__ */
 
+#ifdef __CYGWIN__
+/* the following define is necessary for Cygwin; without it, the
+   Cygwin-supplied ncurses.h sets NCURSES_OPAQUE to 1, and then Python
+   can't get at the WINDOW flags field. */
+#define NCURSES_INTERNALS
+#endif /* __CYGWIN__ */
+
 #ifdef __FreeBSD__
 /*
 ** On FreeBSD, [n]curses.h and stdlib.h/wchar.h use different guards
